network.go

```diff
@@ -237,6 +270,89 @@ func getLinkByName(netHandle *netlink.Handle, name string, expectedLink netlink.
 	return nil, fmt.Errorf("Incorrect link type %s, expecting %s", link.Type(), expectedLink.Type())
 }
 
+func connectVMNetwork(netPair *NetworkInterfacePair) error {
+	switch DefaultNetInterworkingModel {
+	case ModelBridged:
+		return bridgeNetworkPair(netPair)
+	case ModelMacVtap:
+		return tapNetworkPair(netPair)
+	case ModelEnlightened:
+		return fmt.Errorf("Unsupported networking model")
+	default:
+		return fmt.Errorf("Invalid networking model")
+	}
+}
+
+func tapNetworkPair(netPair *NetworkInterfacePair) error {
+	netHandle, err := netlink.NewHandle()
+	if err != nil {
+		return err
+	}
+	defer netHandle.Delete()
+
+	vethLink, err := getLinkByName(netHandle, netPair.VirtIface.Name, &netlink.Veth{})
+	if err != nil {
+		return fmt.Errorf("Could not get veth interface: %s", err)
```

Comment:
Will the error include details from `netPair` or is it worth adding this to the error (same question to the errors returned below)?

Reply:
I will add more details to both the parent error and here. The message does not provide enough debug information otherwise.